### PR TITLE
Training pipeline

### PR DESCRIPTION
**Pull request by pshivraj**
This PR is an end to end pipeline for model Training. It includes:

- Data pre-processing using ```pre_process.py``` to generate .h5 file for masks.

- Custom Config for clomask Dataset. (```config.py```)

- Model Training using coco and imagenet weights. (```train.py```)

This caters to following issues: #18 #17 #26

### CODE REVIEW

**Comment by tejasmhos** [comment]
I think the training pipeline is fine. It can be merged to the master branch.

**Review by tejasmhos — DISMISSED** [review]

File: mrcnn/scripts/train.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            self.model.load_weights(weights_path, by_name=True, exclude=["mrcnn_class_logits", "mrcnn_bbox_fc",
+                                                                   "mrcnn_bbox", "mrcnn_mask"])
+        elif self.init_weights == "coco":
+            weights_path = COCO_PATH
+            self.model.load_weights(weights_path, by_name=True, exclude=["mrcnn_class_logits", "mrcnn_bbox_fc",
+                                                                   "mrcnn_bbox", "mrcnn_mask"])
+
+    def model_train(self):
+
+        augmentation = iaa.Sequential([
```
Comment (lmtoan) [inline mrcnn/scripts/train.py:150]: Keep this line number and file in mind whenever Yipeng asks if we perform any image augmentation during training.

**Comment by lmtoan** [comment]
Thanks Puru. Since we need to train images with no bottle/box/bag, should we stash more commit on this branch or merge this and create another feature branch? 

Just cosmetics:
- Now that we move the `mrcnn/` module to our repository, let's add `__init__.py`, `setup.py`, and other `requirements.txt` to establish it as a proper module?
- I think all scripts in `scripts/*` should surface to `mrcnn/` and move the `mask_data/` out of the current folder. I like `mask_data` to be in our repo for testing and documentation.

File: mrcnn/scripts/train.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+class ClomaskDataset(utils.Dataset):
+    """
+        Class wrapper for clomask dataset.
+    """
+
+    def load_shapes(self, id_list, train_path):
+        """
+           Initialize the class with dataset info.
+        """
+        with open(ROOT_DIR + 'id_map.json') as mapping_file:
```
Comment (lmtoan) [inline mrcnn/scripts/train.py:23]: Could you check in the `id_map.json` file? Or I can add that long list of class labels in `config.py` (with "book" becomes "bags").

**Review by havanagrawal — DISMISSED** [review]

File: mrcnn/scripts/train.py (hunk 2 above)
Comment (pshivraj) [reply mrcnn/scripts/train.py:23]: @lmtoan I have added the ```id_map.json``` to the pull request. Can you approve my latest commit so that I can merge.

**Comment by pshivraj** [comment]
I am closing this pull request for now.
